code/defines/obj/weapon.dm

```diff
@@ -390,50 +390,128 @@
 			else
 				user.show_message("<span class='rose'>There are already two weights on this [src]!</span>")
 
-/obj/item/weapon/legcuffs/beartrap
+/datum/locking_category/beartrap
+		flags = LOCKED_CAN_LIE_AND_STAND
+
+/obj/item/weapon/beartrap
 	name = "bear trap"
 	throw_speed = 2
 	throw_range = 1
-	layer = OPEN_DOOR_LAYER
+	siemens_coefficient = 1
+	plane = OBJ_PLANE
+	layer = BELOW_CLOSED_DOOR_LAYER
+	icon = 'icons/obj/items.dmi'
 	icon_state = "beartrap0"
 	desc = "A trap used to catch bears and other legged creatures."
+	flags = FPRINT
+	origin_tech = Tc_MATERIALS + "=1"
 	starting_materials = list(MAT_IRON = 50000)
 	w_type = RECYK_METAL
+	w_class = W_CLASS_LARGE
+	anchored = FALSE
+	var/breakouttime = 60
 	var/armed = 0
+	var/trapped = 0
+	var/datum/organ/external/trappedorgan = "LIMB_LEFT_LEG"
+	var/mob/living/trappeduser
 	var/obj/item/weapon/grenade/iedcasing/IED = null
 	var/image/ied_overlay
 
-/obj/item/weapon/legcuffs/beartrap/New()
+/obj/item/weapon/beartrap/New()
 	..()
 	ied_overlay = image('icons/obj/items.dmi')
 	ied_overlay.icon_state = "beartrap_ied"
 
-/obj/item/weapon/legcuffs/beartrap/armed
+/obj/item/weapon/beartrap/armed
 	armed = 1
+	anchored = TRUE
 	icon_state = "beartrap1"
 
-/obj/item/weapon/legcuffs/beartrap/suicide_act(mob/user)
-	to_chat(viewers(user), "<span class='danger'>[user] is putting the [src.name] on \his head! It looks like \he's trying to commit suicide.</span>")
-	return (SUICIDE_ACT_BRUTELOSS)
-
-/obj/item/weapon/legcuffs/beartrap/update_icon()
+/obj/item/weapon/beartrap/suicide_act(mob/user as mob)
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		var/datum/organ/external/head/head_organ = H.get_organ(LIMB_HEAD)
+		if(head_organ)
+			to_chat(viewers(user), "<span class='danger'>[user] is arming the [src.name] on \his head! It looks like \he's trying to commit suicide!</span>")
+			playsound(src, 'sound/effects/snap.ogg', 75, 1)
+			head_organ.explode()
+			return (SUICIDE_ACT_BRUTELOSS)
+
+/obj/item/weapon/beartrap/update_icon()
 	icon_state = "beartrap[armed]"
 
-/obj/item/weapon/legcuffs/beartrap/attack_self(mob/user as mob)
+/obj/item/weapon/beartrap/attack_self(mob/user as mob)
 	..()
 	if(ishuman(user) && !user.stat && !user.restrained())
 		armed = !armed
 
 		update_icon()
 
-		to_chat(user, "<span class='notice'>[src] is now [armed ? "armed" : "disarmed"]</span>")
+		to_chat(user, "<span class='notice'>The [src.name] is now [armed ? "armed" : "disarmed"]</span>")
```

Comment:
This is fine, but for future reference you can also do \The [src] instead of The [src.name]

Reply:
Will change, thanks

Reply:
(he didn't change it)